src/Software.php

```diff
@@ -78,6 +78,14 @@ public static function getMenuShorcut()
         return 's';
     }
 
+    public function isDynamic()
+    {
+        if ($this->fields['is_dynamic'] != null && $this->fields['id'] != 0) {
```

Comment:
```suggestion
        if (isset($this->fields['is_dynamic']) && $this->fields['is_dynamic'] && !$this->isNewItem()) {
```

Reply:
I'm not sure base CommonDBTM would not be enough. Why an override is required?

Reply:
Indeed, That was before we added the base field